db/migrate/20211012145201_add_services_column_to_newbookings.rb

```diff
@@ -0,0 +1,7 @@
+class AddServicesColumnToNewbookings < ActiveRecord::Migration[6.1]
+  def change
+    add_column :newbookings, :service_id, :integer
```

Comment:
If I check out this code and run these migrations, it will fail.

`rails db:migrate` runs all of the migrations one after other. 

Can you tell me why it will fail?

Reply:
ok. First.... there is no shame in not knowing and it is much better to say you're not sure than say "yes" and lie 😄 

line 3 (above) creates a new column called `service_id` on the newbookings table
line 4 of this file inserts a value for a service_id (1) into the newbookings table.
line 5 then adds a foreign key constraint that says "all of the values for service id must exist in the services table"

If I run `rails db:migrate` locally, the file above this one will be executed first, then this immediately after. The file above is creating the `services` table.

What that results in is:
1. create the services table
2. add a service id column to newbookings
3. make all of the newbookings point to service with ID 1
4. tell the database "if the service ID in newbookings doesn't exist, explode!"

When I get to item 4 on this list.... there is no service with ID 1.... 💥 

So... knowing that, any ideas on how to go about fixing it?

Reply:
Create a service using SQL during the migration?

Reply:
I'm fairly certain I have now fixed this.